src/java.base/share/classes/java/lang/reflect/Proxy.java

```diff
@@ -1245,13 +1246,13 @@ public static InvocationHandler getInvocationHandler(Object proxy)
      *         <li>the given {@code method} is overridden directly or indirectly by
      *             the proxy interfaces and the method reference to the named
      *             method never resolves to the given {@code method}; or</li>
-     *         <li>if length of given {@code args} array doesn't match the number of
-     *             parameters of the method to be invoked or if {@code args} is null
-     *             and the method to be invoked has parameters</li>
-     *         <li>if any of the {@code args} elements can't be assigned to the
-     *             boxed type of the corresponding method parameter or any of the
-     *             {@code args} elements is null while the corresponding method
-     *             parameter is of primitive type</li>
+     *         <li>the length of the given {@code args} array does not match the
+     *             number of parameters of the method to be invoked; or</li>
+     *         <li>any of the {@code args} elements fails the unboxing
+     *             conversion if the corresponding method parameter type is
+     *             a primitive type; or if, after possible unboxing, any of the
+     *             {@code args} elements cannot be assigned to the corresponding
+     *             method parameter type.</li>
      *         </ul>
```

Comment:
After trying it in practice, I see your words are correct. 1st unboxing happens and then assignment.